jdaviz/core/helpers.py

```diff
@@ -42,6 +43,17 @@ def __init__(self, app=None, verbosity='info'):
             self.app = app
         self.app.verbosity = verbosity
 
+        self.app.hub.subscribe(self, SubsetCreateMessage,
+                               handler=self._on_subset_create)
+
+    def _on_subset_create(self, msg):
+        # viewers don't have access to the app/hub to subscribe to messages, so we'll
```

Comment:
🤯

Reply:
I wonder if this would be as simple as having our viewer classes subclass from `HubListener` like the helper classes. I'll investigate...

Reply:
that's an interesting idea that might work!  I managed to get it working with changes to glue-jupyter, but figured that wasn't worth it quite yet.  Any luck?

Reply:
I got distracted after remembering that it's not quite as simple as just subclassing that, the viewer also needs access to the `app` object. Haven't had much time to hack at it yet.

Reply:
My quick attempts were unsuccessful (inheriting from `HubListener` and then subscribing via `self.session.hub`)...

Reply:
Yeah, you need `app.session.hub`, and the viewers don't have a hook back up to the app as far as I can remember.

Reply:
FWIW: For the Compass work, I got to the point of hacking `TemplateMixin` to accept viewer as `app` but that didn't work either. 🐰 🕳️